src/OpenColorIO/ops/fixedfunction/FixedFunctionOpGPU.cpp

```diff
@@ -347,6 +349,1305 @@ void Add_GamutComp_13_Inv_Shader(GpuShaderText & ss,
+
+    ss.newLine() << "while (i_lo + 1 < i_hi)";
+    ss.newLine() << "{";
+    ss.indent();
+
+    ss.newLine() << ss.floatDecl("hcur") << " = " << hues_array_name << "[i];";
+
+    ss.newLine() << "if (h > hcur)";
+    ss.newLine() << "{";
+    ss.indent();
+    ss.newLine() << "i_lo = i;";
+    ss.dedent();
+    ss.newLine() << "}";
+    ss.newLine() << "else";
+    ss.newLine() << "{";
+    ss.indent();
+    ss.newLine() << "i_hi = i;";
+    ss.dedent();
+    ss.newLine() << "}";
+    ss.newLine() << "i = " << ss.intKeyword() << "((i_lo + i_hi) / 2.0);";
+
+    ss.dedent();
+    ss.newLine() << "}";
+
+    if (dimensions == GpuShaderDesc::TEXTURE_1D)
+    {
+        ss.newLine() << ss.float3Decl("lo") << " = " << ss.sampleTex1D(name, std::string("(i_hi - 1 + 0.5) / ") + std::to_string(g.gamut_cusp_table.total_size)) << ".rgb;";
+        ss.newLine() << ss.float3Decl("hi") << " = " << ss.sampleTex1D(name, std::string("(i_hi + 0.5) / ") + std::to_string(g.gamut_cusp_table.total_size)) << ".rgb;";
+    }
+    else
+    {
+        ss.newLine() << ss.float3Decl("lo") << " = " << ss.sampleTex2D(name, ss.float2Const(std::string("(i_hi - 1 + 0.5) / ") + std::to_string(g.gamut_cusp_table.total_size), "0.5")) << ".rgb;";
+        ss.newLine() << ss.float3Decl("hi") << " = " << ss.sampleTex2D(name, ss.float2Const(std::string("(i_hi + 0.5) / ") + std::to_string(g.gamut_cusp_table.total_size), "0.5")) << ".rgb;";
+    }
+
+    ss.newLine() << ss.floatDecl("t") << " = (h - lo.b) / (hi.b - lo.b);";
+    ss.newLine() << ss.floatDecl("cuspJ") << " = " << ss.lerp("lo.r", "hi.r", "t") << ";";
+    ss.newLine() << ss.floatDecl("cuspM") << " = " << ss.lerp("lo.g", "hi.g", "t") << ";";
+
+    ss.newLine() << "return " << ss.float2Const("cuspJ", "cuspM") << ";";
+
+    ss.dedent();
+    ss.newLine() << "}";
+
+    shaderCreator->addToHelperShaderCode(ss.string().c_str());
+
+    return name;
+}
+
+std::string _Add_Gamma_table(
+    GpuShaderCreatorRcPtr & shaderCreator,
+    unsigned resourceIndex,
+    const ACES2::GamutCompressParams & g)
+{
+    // Reserve name
+    std::ostringstream resName;
+    resName << shaderCreator->getResourcePrefix()
+            << std::string("_")
+            << std::string("upper_hull_gamma_table")
+            << resourceIndex;
+
+    // Note: Remove potentially problematic double underscores from GLSL resource names.
+    std::string name(resName.str());
+    StringUtils::ReplaceInPlace(name, "__", "_");
+
+    // Register texture
+    GpuShaderDesc::TextureDimensions dimensions = GpuShaderDesc::TEXTURE_1D;
+    if (shaderCreator->getLanguage() == GPU_LANGUAGE_GLSL_ES_1_0
+        || shaderCreator->getLanguage() == GPU_LANGUAGE_GLSL_ES_3_0
+        || !shaderCreator->getAllowTexture1D())
+    {
+        dimensions = GpuShaderDesc::TEXTURE_2D;
+    }
+
+    shaderCreator->addTexture(
+        name.c_str(),
+        GpuShaderText::getSamplerName(name).c_str(),
+        g.gamut_cusp_table.total_size,
```

Comment:
Minor todo fix for later: same value, but uses `gamut_cusp_table` instead of `upper_hull_gamma_table`.